### DISCUSSION TITLE
Can dify support using knowledge base via  Annual Report or  Annual Document of many different year

### DISCUSSION THREAD

**Discussion by BingGo-1** [langgenius/dify#6226] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I was tring to use Knowledge Base  for a inner project.  At the beginning, everything was going as i planed. But  recently i found a problem and i can not fix it. I want to upload some special documents into Knowledge，the special documents means they are very similar，like Annual Report of 2021，Annual Report of 2022, and so on.  At this situation, when i ask a question about a year, many documents of different year will be found and returned. It seems like it can not recognize the specific year because the reports do not contain the year's info. Now i want the only document of the specific year could be found and returned, can there be some approaches to study and try or this can not be supported? Thanks very much!

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by mboo2005** [comment]
Any progress?

**Comment by mboo2005** [comment]
Does dify have any plan to support structured data ? If not I plan to develope the feature
